scripts/serve_api.py

```diff
@@ -0,0 +1,71 @@
+"""Serve SourceCred API, for consumption by Observable notebooks.
+
+Usage:
+
+    python3 ./scripts/serve_api.py [PORT]
+
+from the SourceCred repository root. Omit the port to try a default, or
+specify port 0 to pick a free port automatically.
+
+This server binds to `localhost`, which is a loopback address only
+accessible from your machine, not the local network.
+
+Corresponding Observable code:
+
+    sourcecred = {
+      const server = "http://localhost:9009";
+      const blob = await fetch(server).then((r) => r.blob());
```

Comment:
Just chiming in here: don't forget to catch the await for an error!
```
await fetch(server)
  .then((r) => r.blob())
  .catch((e) => {
    throw new Error(e);
  })
```

Reply:
I'm all for error handling, but is there a benefit to throwing the error again vs letting it go as an unhandled error?

Reply:
Unhandled errors will at some point be deprecated by Node. Potentially with 14. And since this is just a suggestion, my feeling is that its best to lead by example.

Further, it allows you to swallow the error and do something else than merely throwing an unhandled error as an exception.

Reply:
> Unhandled errors will at some point be deprecated by Node. Potentially with 14.

Interesting! Hopefully one day the stock Promises will be as useful as something like https://github.com/fluture-js/Fluture.

Had forced error handling since inception. And also makes a distinction between expected (a rejection value) and unexpected errors (divide by zero in your map function).

Reply:
Isn’t `p.catch((e) => { throw new Error(e); })` exactly the same as `p`?
Errors thrown from within promise callbacks become rejections. I’m not
sure what you’re suggesting.

Recall also that this code is executed by Observable, not Node. If there
is an error, it will be clearly displayed in the UI, and the user will
debug it. There is nothing useful to do by handling the error.